### PR TITLE
Update links to Kubernetes documentation

### PR DESCRIPTION
**Pull request by ghost**
This is a follow-up to issue https://github.com/openshift/openshift-docs/issues/9979.

@openshift/team-documentation PTAL. Changes made to fix broken links to Kubernetes docs in GitHub.

### CODE REVIEW

**Comment by tmorriso-rh** [comment]
This issue also affects 3.1, 3.2, and 3.3.

**Comment by tmorriso-rh** [comment]
@openshift/team-documentation PTAL. Thanks!

File: architecture/core_concepts/pods_and_services.adoc [hunk 1]
```diff
@@ -35,17 +35,16 @@ OpenShift treats pods as largely immutable; changes cannot be made to
 a pod definition while it is running. OpenShift implements changes by
 terminating an existing pod and recreating it with modified configuration,
 base image(s), or both. Pods are also treated as expendable, and do not
-maintain state when recreated. Therefore pods should usually be managed by
+maintain state when recreated. Therefore, pods should usually be managed by
```
Comment (kalexand-rh) [inline architecture/core_concepts/pods_and_services.adoc]: maybe s/pods should usually be managed by/manage pods with

File: architecture/core_concepts/pods_and_services.adoc [hunk 2]
```diff
@@ -35,17 +35,16 @@ OpenShift treats pods as largely immutable; changes cannot be made to
...
 higher-level xref:deployments.adoc#replication-controllers[controllers],
 rather than directly by users.
 
 Below is an example definition of a pod that provides a long-running
 service, which is actually a part of the OpenShift infrastructure: the
 private Docker registry. It demonstrates many features of pods, most of
-which are discussed in other topics and thus only briefly mentioned here:
+which are discussed in other topics and thus only briefly mentioned here.
```
Comment (kalexand-rh) [inline architecture/core_concepts/pods_and_services.adoc]: I'd consider adding a link to one of these topics.

File: architecture/core_concepts/pods_and_services.adoc [hunk 3]
```diff
@@ -97,16 +96,13 @@ spec:
     secret:
       secretName: default-token-br6yz
 ----
-
-====
-
 <1> Pods can be "tagged" with one or more xref:labels[labels], which can then
 be used to select and manage groups of pods in a single operation. The labels
-are stored in key/value format in the `*metadata*` hash. One label in this
+are stored in key/value format in the `metadata` hash. One label in this
```
Comment (kalexand-rh) [inline architecture/core_concepts/pods_and_services.adoc]: s/"key/value"/key-value

File: architecture/core_concepts/pods_and_services.adoc [hunk 4]
```diff
@@ -97,16 +96,13 @@ spec:
...
 example is *docker-registry=default*.
 <2> Pods must have a unique name within their
 xref:projects_and_users.adoc#namespaces[namespace]. A pod definition may specify
-the basis of a name with the `*generateName*` attribute, and random characters
+the basis of a name with the `generateName` attribute, and random characters
```
Comment (kalexand-rh) [inline architecture/core_concepts/pods_and_services.adoc]: no comma

File: architecture/core_concepts/pods_and_services.adoc [hunk 5]
```diff
@@ -141,15 +137,14 @@ are filled by OpenShift automatically after the pod is created and
 its lifecycle begins. The
 xref:../../rest_api/kubernetes_v1.adoc#rest-api-kubernetes-v1[Kubernetes API documentation]
 has complete details of the pod REST API object attributes, and the
-link:https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/user-guide/pods.md[Kubernetes pod documentation]
+link:https://github.com/kubernetes/kubernetes/blob/release-1.0/docs/user-guide/pods.md[Kubernetes pod documentation]
```
Comment (kalexand-rh) [inline architecture/core_concepts/pods_and_services.adoc]: Is release-1.0 valid for all of these collections? Shouldn't it be different for each OCP version? (All places)

**Review by kalexand-rh — COMMENTED** [review]
I have a couple of nitpicks and questions.

**Pull request merged**